open-craft/frontend-app-authn
# feat: Add plugin slot for login page

Backport of openedx/frontend-app-authn#1354 to Redwood.

## Description

This change adds a plugin slot for the login page allowing it to be customised.

Since there was a dependency conflict between frontend-plugin-framework and the react-hooks testing package, the react-hooks testing package has been removed and a replaced with a simple mechanism for testing hooks.

Since this touched the Login Page those have also been refactored to move away from redux connect.

## Supporting information

openedx/frontend-app-authn#1354

## Testing instructions

1. Use env.config.jsx below.

    ```
    import { PLUGIN_OPERATIONS } from '@openedx/frontend-plugin-framework';
    
    
    const config = {
        pluginSlots: {
            login_page_slot: {
                keepDefault: true,
                plugins: [
                    {
                        op: PLUGIN_OPERATIONS.Wrap,
                        widgetId: 'default_contents',
                        wrapper: ({ component }) => {
                            return (
                                <div>
                                    <span>Hello world</span>
                                    {component}
                                </div>
                            );
                        },
                    },
                ],
            },
        },
    };
    
    export default config;
    ```
3. See "Hello world" text before login form

## Deadline

None

## Other information

Private-ref: https://tasks.opencraft.com/browse/BB-7742
